snap/snapenv/snapenv.go

```diff
@@ -106,8 +106,17 @@ func userEnv(info *snap.Info, home string) map[string]string {
 	return result
 }
 
-// envMap creates a map from the given environment string list, e.g. the
-// list returned from os.Environ()
+var unsafeEnv = map[string]bool{
+	"HOSTALIASES": true,
+	"TMPDIR":      true,
```

Comment:
I'm not sure TMPDIR should be preserved. The designated directory may not exist inside the execution environment. What is your motivation for preserving it?